Modules/StudyProgramme/classes/class.ilObjStudyProgrammeListGUI.php

```diff
@@ -68,6 +75,91 @@ public function getCommandLink(string $cmd): string
         return $this->ctrl->getLinkTargetByClass("ilobjstudyprogrammegui", $cmd);
     }
 
+    public function getAsListItem(int $ref_id, int $obj_id, string $type, string $title, string $description): ?\ILIAS\UI\Component\Item\Item
+    {
+        $this->initItem(
+            $ref_id,
+            $obj_id,
+            $type,
+            $title,
+            $description
+        );
+        $this->insertCommands(true, true);
+
+        $prg = new ilObjStudyProgramme($ref_id);
```

Comment:
Creating new instances is a bad practice for liste items. These are expensive calls (resulting in many database queries). As with the other issue, you just copied that from `getListItemHTML`.
Unfortunately I did not find the reference (README, DEV GUIDE) for this, but somewhere is state: Don't create new `ilObj*` instances, you the `_preloadData` functions in the `ilObj*Access` class and use static caches to retrieve the preloaded values in the `ilObj*ListGUI`.